File: SQL Lookup App/FormMain.vb

Imports SQL_Lookup_App.Intelligence

Public Class FormMain
    Private Sub FormMain_Load(sender As Object, e As EventArgs) Handles Me.Load
        ' Load Saved Settings
        Me.txtConnection.Text = My.Settings.ConnectionString
        Me.txtCommand.Text = My.Settings.CommandText

        ' Set Tab 
        Me.txtResults.SelectionTabs = New Integer() {300, 600, 800, 1000, 1175, 1350, 1525, 1700, 1875, 2050, 2225, 2400, 2575, 2750, 2925, 3100}
    End Sub

    Private Sub FormMain_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        ' Save input details
        My.Settings.ConnectionString = Me.txtConnection.Text
        My.Settings.CommandText = Me.txtCommand.Text
        My.Settings.Save()
    End Sub


    Private Sub CmdExecute_Click(sender As Object, e As EventArgs) Handles cmdExecute.Click
        LockForm(True)
        My.Settings.ConnectionString = Me.txtConnection.Text
        My.Settings.CommandText = Me.txtCommand.Text
        My.Settings.Save()


        If Me.txtConnection.Text.StartsWith("Data Source=(") Then
            ' Oracle
            Using oDB As New OracleSrv(Me.txtConnection.Text)
                AddHandler oDB.ErrorMessage, Sub(category As String, message As String) LogError(category, message)
                AddHandler oDB.ResultMessage, Sub(message As String) LogResult(message)

                oDB.ExecuteStatement(Me.txtCommand.Text)
            End Using
        Else
            ' MS SQL
            Using oDB As New MsSqlSrv(Me.txtConnection.Text)
                AddHandler oDB.ErrorMessage, Sub(category As String, message As String) LogError(category, message)
                AddHandler oDB.ResultMessage, Sub(message As String) LogResult(message)

                oDB.ExecuteStatement(Me.txtCommand.Text)
            End Using
        End If

        LockForm(False)
    End Sub



    Private Delegate Sub LogErrorDelegate(category As String, message As String)
    Private Sub LogError(category As String, message As String)
        If txtResults.InvokeRequired Then
            Dim d = New LogErrorDelegate(AddressOf LogError)
            txtResults.Invoke(d)
        Else
            Me.txtResults.Text &= category & vbTab & message & Environment.NewLine
        End If
        Application.DoEvents()
    End Sub


    Private Delegate Sub LogResultDelegate(message As String)
    Private Sub LogResult(message As String)
        If txtResults.InvokeRequired Then
            Dim d = New LogResultDelegate(AddressOf LogResult)
            txtResults.Invoke(d)
        Else
            Me.txtResults.Text &= message & Environment.NewLine
        End If
        Application.DoEvents()
    End Sub



    Private Sub LockForm(ByVal lockit As Boolean)

        Me.txtConnection.Enabled = Not lockit
        Me.txtCommand.Enabled = Not lockit

        Me.cmdConnectionLoad.Enabled = Not lockit
        Me.cmdExecute.Enabled = Not lockit
        Me.cmdClearResults.Enabled = Not lockit

        Me.Cursor = If(lockit, Cursors.WaitCursor, Cursors.Default)

        Me.Refresh()
        My.Application.DoEvents()
    End Sub


    Private Sub CmdConnectionLoad_Click(sender As Object, e As EventArgs) Handles cmdConnectionLoad.Click

        ' Init box with last used
        Dim oConnection As New FormConnectionDesign() With {
            .DatabaseName = My.Settings.CS_Database,
            .ServerName = My.Settings.CS_Server,
            .Security = My.Settings.CS_Secrurity,
            .UserName = My.Settings.CS_Username,
            .Password = My.Settings.CS_Password,
            .DatabaseType = My.Settings.CS_Type
        }

        'Show Form
        Dim result = oConnection.ShowDialog(owner:=Me)
        If result = DialogResult.OK Then
            ' If good save and create string
            My.Settings.CS_Database = oConnection.DatabaseName.Trim()
            My.Settings.CS_Server = oConnection.ServerName.Trim()
            My.Settings.CS_Secrurity = oConnection.Security
            My.Settings.CS_Username = oConnection.UserName.Trim()
            My.Settings.CS_Password = oConnection.Password.Trim()
            My.Settings.CS_Type = oConnection.DatabaseType
            My.Settings.Save()

            ' Create CS
            Me.txtConnection.Text = CSHelper.GetCS(My.Settings.CS_Type, My.Settings.CS_Server, My.Settings.CS_Database, My.Settings.CS_Secrurity, My.Settings.CS_Username, My.Settings.CS_Password)
        End If

    End Sub

    Private Sub CmdClearResults_Click(sender As Object, e As EventArgs) Handles cmdClearResults.Click
        Me.txtResults.Text = String.Empty
    End Sub

End Class

File: SQLLookupTool/FormMain.cs
using SQLLookupTool.Intelligence;
using SQLLookupTool.Properties;
using System;
using System.Windows.Forms;

namespace SQLLookupTool
{
    public partial class FormMain : Form
    {
        public FormMain()
        {
            InitializeComponent();


            // Load Saved Settings
            this.txtConnection.Text = Settings.Default.ConnectionString;
            this.txtCommand.Text = Settings.Default.CommandText;

            // Set Tab Stops
            this.txtResults.SelectionTabs = new int[] { 300, 600, 800, 1000, 1175, 1350, 1525, 1700, 1875, 2050, 2225, 2400, 2575, 2750, 2925, 3100 };

        }

        private void CmdConnectionLoad_Click(object sender, EventArgs e)
        {
            // Init box with last used
            var oConnection = new FormConnectionDesign();

            // Show Form
            DialogResult result = oConnection.ShowDialog(owner: this);
            if (result == DialogResult.OK && !string.IsNullOrWhiteSpace(oConnection.ServerName))
            {
                // If good save and create string
                string CS_Database = oConnection.DatabaseName;
                string CS_Server = oConnection.ServerName;
                bool CS_Secrurity = oConnection.Security;
                string CS_Username = oConnection.UserName;
                string CS_Password = oConnection.Password;
                CSTypeList CS_Type = oConnection.DatabaseType;

                // Create CS
                this.txtConnection.Text = CSHelper.GetConnectionString(CS_Type, CS_Server, CS_Database, CS_Secrurity, CS_Username, CS_Password);
            }
        }


        private void CmdExecute_Click(object sender, EventArgs e)
        {
            // Lock form and save settings
            LockForm(true);
            Settings.Default.ConnectionString = this.txtConnection.Text;
            Settings.Default.CommandText = this.txtCommand.Text;
            Settings.Default.Save();


            if (this.txtConnection.Text.StartsWith("Data Source=("))
            {
                // Oracle
                using (OracleServer oDB = new OracleServer(this.txtConnection.Text))
                {
                    oDB.ErrorMessage += (category, message) => LogError(category, message);
                    oDB.ResultMessage += message => LogResult(message);

                    oDB.ExecuteStatement(this.txtCommand.Text);
                }
            }
            else
                // MS SQL
                using (MsSqlServer oDB = new MsSqlServer(this.txtConnection.Text))
                {
                    oDB.ErrorMessage += (category, message) => LogError(category, message);
                    oDB.ResultMessage += message => LogResult(message);

                    oDB.ExecuteStatement(this.txtCommand.Text);
                }

            LockForm(false);
        }



        private void CmdClearResults_Click(object sender, EventArgs e)
        {
            this.txtResults.Text = String.Empty;
        }



        private void LockForm(bool lockit)
        {
            this.txtConnection.Enabled = !lockit;
            this.txtCommand.Enabled = !lockit;

            this.cmdConnectionLoad.Enabled = !lockit;
            this.cmdExecute.Enabled = !lockit;
            this.cmdClearResults.Enabled = !lockit;

            this.Cursor = lockit ? Cursors.WaitCursor : Cursors.Default;

            this.Refresh();
        }


        private delegate void LogErrorDelegate(string category, string message);
        private void LogError(string category, string message)
        {
            if (this.txtResults.InvokeRequired)
            {
                var d = new LogErrorDelegate(LogError);
                this.txtResults.Invoke(d);
            }
            else
                this.txtResults.Text += category + "\t" + message + Environment.NewLine;
            Application.DoEvents();
        }


        private delegate void LogResultDelegate(string message);
        private void LogResult(string message)
        {
            if (this.txtResults.InvokeRequired)
            {
                var d = new LogResultDelegate(LogResult);
                this.txtResults.Invoke(d);
            }
            else
                this.txtResults.Text += message + Environment.NewLine;
            Application.DoEvents();
        }
    }
}
